src/client/src/apps/MainRoute/Router.tsx

```diff
@@ -1,60 +1,97 @@
-import React, { FC } from 'react'
+import React, { FC, useEffect } from 'react'
 import { Route, Switch } from 'react-router-dom'
 import { RouteWrapper } from 'rt-components'
 import { OpenFinWindowFrame, OpenFinSubWindowFrame } from 'rt-platforms/openfin-platform/components'
 import { StatusDisplayContainer } from 'rt-platforms/openfin-platform/components/OpenFinStatusConnection/StatusContainers'
 import { currencyFormatter } from 'rt-util'
 import { AnalyticsRoute, BlotterRoute, SpotRoute, ShellRoute, TileRoute } from './routes'
+import { useDispatch, useSelector } from 'react-redux'
+import { Trade } from 'rt-types'
+import { GlobalState } from 'StoreTypes'
 import { OpenFinContactDisplay } from 'rt-platforms/openfin-platform/components/OpenFinContactButton'
+import { BlotterActions } from './widgets/blotter/actions'
 
-export const Router: FC = () => (
-  <Switch>
-    <Route
-      path="/analytics"
-      render={() => (
-        <RouteWrapper windowType="sub" title="Analytics">
-          <AnalyticsRoute />
-        </RouteWrapper>
-      )}
-    />
-    <Route
-      path="/blotter"
-      render={routeProps => (
-        <RouteWrapper windowType="sub" title="Trades">
-          <BlotterRoute {...routeProps} />
-        </RouteWrapper>
-      )}
-    />
-
-    <Route
-      path="/tiles"
-      render={() => (
-        <RouteWrapper windowType="sub" title="Live Rates">
-          <TileRoute />
-        </RouteWrapper>
-      )}
-    />
-    <Route
-      path="/spot/:symbol"
-      render={routeProps => (
-        <RouteWrapper windowType="sub" title={currencyFormatter(routeProps.match.params.symbol)}>
-          <SpotRoute {...routeProps} />
-        </RouteWrapper>
-      )}
-    />
-    <Route
-      exact
-      path="/"
-      render={() => (
-        <RouteWrapper>
-          <ShellRoute />
-        </RouteWrapper>
-      )}
-    />
-
-    <Route path="/openfin-window-frame" render={() => <OpenFinWindowFrame />} />
-    <Route path="/openfin-sub-window-frame" render={() => <OpenFinSubWindowFrame />} />
-    <Route path="/status" render={() => <StatusDisplayContainer />}/>
-    <Route path="/contact" render={() => <OpenFinContactDisplay />}/>
-  </Switch>
-)
+const { highlightTradeAction } = BlotterActions
+
+export const Router: FC = () => {
+  const dispatch = useDispatch()
+  const switchHighlight = (trade: Trade, on: boolean) => ({
+    ...trade,
+    highlight: on,
+  })
+
+  const trades = useSelector((state: GlobalState) => state.blotterService.trades)
```

Comment:
Can highlights be moved to be under the tiles route?

Reply:
I had that same thought, but it seems when the eventListener is only bound to a specific route, the SW message doesn't propagate the way you might expect it to, and the highlight doesn't work.

(There might be a way to do it, but it wouldn't be as simple as just moving over the code there, and I'm not sure what changes would have to be made to get it done).

Reply:
I agree with @stewart-adaptive - this really clutters up the Router file. What I would suggest is if there isn't a clear way to get the highlighting to work for now, it is put on ice at the moment, and updating the design of the notifications (new icon, text, etc.) takes priority.